server/handlers.go

```diff
@@ -52,80 +56,502 @@ func handleRequestDownloaderMissing(logger *zap.Logger) http.HandlerFunc {
 	}
 }
 
-func handleAccountImages(multiaccountsDB *multiaccounts.Database, logger *zap.Logger) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		params := r.URL.Query()
+type ParsedParams struct {
+	KeyUID          string
+	PublicKey       string
+	ImageName       string
+	ImagePath       string
+	FullName        string
+	InitialsLength  int
+	FontFile        string
+	FontSize        float64
+	Color           color.Color
+	BgSize          int
+	BgColor         color.Color
+	UppercaseRatio  float64
+	Theme           ring.Theme
+	Ring            bool
+	IndicatorSize   int
+	IndicatorBorder int
+	IndicatorColor  color.Color
+
+	AuthorID     string
+	URL          string
+	MessageID    string
+	AttachmentID string
+
+	Hash     string
+	Download bool
+}
 
-		keyUids, ok := params["keyUid"]
-		if !ok || len(keyUids) == 0 {
-			logger.Error("no keyUid")
-			return
+func ParseParams(logger *zap.Logger, params url.Values) ParsedParams {
+	parsed := ParsedParams{}
+	parsed.Color = color.Transparent
+	parsed.BgColor = color.Transparent
+	parsed.IndicatorColor = color.Transparent
+	parsed.UppercaseRatio = 1.0
+
+	keyUids := params["keyUid"]
+	if len(keyUids) != 0 {
+		parsed.KeyUID = keyUids[0]
+	}
+
+	pks := params["publicKey"]
+	if len(pks) != 0 {
+		parsed.PublicKey = pks[0]
+	}
+
+	imageNames := params["imageName"]
+	if len(imageNames) != 0 {
+		if filepath.IsAbs(imageNames[0]) {
+			if _, err := os.Stat(imageNames[0]); err == nil {
+				parsed.ImagePath = imageNames[0]
+			} else {
+				logger.Error("ParseParams: image not exit", zap.String("imageName", imageNames[0]))
+				return parsed
+			}
+		} else {
+			parsed.ImageName = imageNames[0]
 		}
-		imageNames, ok := params["imageName"]
-		if !ok || len(imageNames) == 0 {
-			logger.Error("no imageName")
-			return
+	}
+
+	names := params["name"]
+	if len(names) != 0 {
+		parsed.FullName = names[0]
+	}
+
+	parsed.InitialsLength = 2
+	amountInitialsStr := params["length"]
+	if len(amountInitialsStr) != 0 {
+		amountInitials, err := strconv.Atoi(amountInitialsStr[0])
+		if err != nil {
+			logger.Error("ParseParams: invalid initials length")
+			return parsed
+		}
+		parsed.InitialsLength = amountInitials
+	}
+
+	fontFiles := params["fontFile"]
+	if len(fontFiles) != 0 {
+		if _, err := os.Stat(fontFiles[0]); err == nil {
+			parsed.FontFile = fontFiles[0]
+		} else {
+			logger.Error("ParseParams: font file not exit", zap.String("FontFile", fontFiles[0]))
```

Comment:
I think we shouldn't hide the original error when log, https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go

Reply:
fixed, thanks for the info